alibi_detect/base.py

```diff
@@ -229,6 +225,17 @@ def from_config(self): ...
     def _set_config(self): ...
 
 
+@runtime_checkable
+class StatefulDetector(ConfigurableDetector, Protocol):
```

Comment:
Are you planning to add an `OfflineStatefulDetector` in this [PR](https://github.com/ascillitoe/alibi-detect/pull/44/files) out of interest?

Reply:
Good question. I didn't think too much about this. I think whether we need `OfflineStatefulDetector` or not in the future will depend on whether all detectors end up having "offline state" (in which case we won't need the extra protocol to distinguish those that don't) or not. Maybe to be safe `StatefulDetector` should be renamed `OnlineStatefulDetector` for now?

Reply:
It probably makes sense to do so in the future PR so that it makes sense contextually.I'm happy

Reply:
I've renamed it to `OnlineStatefulDetector`. Hope that is OK! :)